marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/managers/MapSearchManager.java

```diff
@@ -117,7 +117,7 @@ public StringHandle search(MapSearchQuery mapSearchQuery) {
 
     public JsonNode sjsSearch(SJSSearchQuery SJSSearchQuery) {
         if (SJSSearchQuery.sourceQuery == null && SJSSearchQuery.database == null) {
-            throw new BadRequestException("Query requires database or sourceQuery");
+            throw new DataHubException("Query requires database or sourceQuery");
```

Comment:
How are we differentiating from a 400 Bad Request and a 500 internal server error in this case?

Reply:
The BadRequestException wasn't actually doing anything - it wasn't capturing a 400, and it wasn't being handled by CustomExceptionHandler. It was also extending Spring Security's AuthenticationException, but it's not an authentication exception. So the class effectively wasn't doing anything, so I removed it and switched this over to the generic DataHubException.  That exception isn't really do anything either - CustomExceptionHandler handles it in its generic exception handler - but it's better than an exception that wasn't doing what its name implies.